raiden-cli/src/routes/payments.ts

```diff
@@ -1,16 +1,73 @@
-import { Router, Request, Response } from 'express';
+import { Router, Request, Response, NextFunction } from 'express';
+import { ErrorCodes, RaidenError } from 'raiden-ts';
+import { timer } from 'rxjs';
+import { toArray, takeUntil } from 'rxjs/operators';
 import { Cli } from '../types';
+import { validateAddressParameter, isInvalidParameterError } from '../utils/validation';
+import { transformSdkTransferToApiPayment } from '../utils/payments';
+
+function isConflictError(error: RaidenError): boolean {
+  return [ErrorCodes.PFS_NO_ROUTES_BETWEEN_NODES].includes(error.message);
+}
+
+async function getPaymentsForTokenAndTarget(this: Cli, request: Request, response: Response) {
+  const allTransfers = await this.raiden.transfers$
+    .pipe(takeUntil(timer(0)), toArray())
+    .toPromise();
+  const filteredTransfers = allTransfers.filter(
+    (transfer) =>
+      transfer.token === request.params.tokenAddress &&
+      transfer.target === request.params.targetAddress,
+  );
+  const payments = filteredTransfers.map((transfer) => transformSdkTransferToApiPayment(transfer));
+  response.json(payments);
+}
+
+async function doTransfer(this: Cli, request: Request, response: Response, next: NextFunction) {
+  try {
+    // TODO: We ignore the provided `lock_timeout` until #1710 provides a better solution
+    const transferKey = await this.raiden.transfer(
+      request.params.tokenAddress,
+      request.params.targetAddress,
+      request.body.amount,
+      { paymentId: request.body.identifier },
+    );
+    this.log.info(`Transfer key: ${transferKey}`);
+    await this.raiden.waitTransfer(transferKey);
+    const allTransfers = await this.raiden.transfers$
+      .pipe(takeUntil(timer(0)), toArray())
+      .toPromise();
+    const newTransfer = allTransfers.filter((transfer) => transfer.key === transferKey)[0];
```

Comment:
Nitpick: It should be safe and more performant to, here, instead of creating an array with all transfers and then filtering, since subscribing to the `transfers$` observable will output the latest state for each transfer, to do it like:
```suggestion
    const newTransfer = await this.raiden.transfers$
      .pipe(first(({ key }) => key === transferKey))
      .toPromise();
```

Reply:
:heavy_check_mark: